instrumentor/controllers/instrumentationdevice/workload_controllers.go

```diff
@@ -37,16 +39,26 @@ type StatefulSetReconciler struct {
 func (r *StatefulSetReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	instrumentedAppName := workload.CalculateWorkloadRuntimeObjectName(req.Name, workload.WorkloadKindStatefulSet)
 	err := reconcileSingleInstrumentedApplicationByName(ctx, r.Client, instrumentedAppName, req.Namespace)
-	return ctrl.Result{}, err
+	return utils.RetryOnConflict(err)
 }
 
 func reconcileSingleInstrumentedApplicationByName(ctx context.Context, k8sClient client.Client, instrumentedAppName string, namespace string) error {
 	var instrumentedApplication odigosv1.InstrumentedApplication
 	err := k8sClient.Get(ctx, types.NamespacedName{Name: instrumentedAppName, Namespace: namespace}, &instrumentedApplication)
 	if err != nil {
-		// changes in workload when there is no instrumented application is not interesting
-		return client.IgnoreNotFound(err)
+		if apierrors.IsNotFound(err) {
+			// if there is no instrumented application, make sure the device is removed from the workload pod template manifest
+			workloadName, workloadKind, err := workload.ExtractWorkloadInfoFromRuntimeObjectName(instrumentedAppName)
+			if err != nil {
+				return err
+			}
+			err = removeInstrumentationDeviceFromWorkload(ctx, k8sClient, namespace, workloadKind, workloadName, ApplyInstrumentationDeviceReasonNoRuntimeDetails)
+			return err
```

Comment:
this means that a conflict error will be printed by the controller runtime - right?